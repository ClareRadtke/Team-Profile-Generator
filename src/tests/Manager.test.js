const { Manager } = require("../lib/classes");

describe(Manager, () => {
  it("should create a new Manager instance", () => {
    const manager = new Manager("John Doe", "123", "foo@example.com", "100");
    expect(manager).toMatchSnapshot();
  });
  describe("getName()", () => {
    it("should return the name", () => {
      const manager = new Manager("John Doe", "123", "foo@example.com", "100");
      expect(manager.getName()).toBe("John Doe");
    });
  });
  describe("getId()", () => {
    it("should return the id", () => {
      const manager = new Manager("John Doe", "123", "foo@example.com", "100");
      expect(manager.getId()).toBe("123");
    });
  });
  describe("getEmail()", () => {
    it("should return the email", () => {
      const manager = new Manager("John Doe", "123", "foo@example.com", "100");
      expect(manager.getEmail()).toBe("foo@example.com");
    });
  });
  describe("getRole()", () => {
    it("should return the role", () => {
      const manager = new Manager("John Doe", "123", "foo@example.com", "100");
      expect(manager.getRole()).toBe("Manager");
    });
  });
  describe("getOfficeNumber()", () => {
    it("should return the office number", () => {
      const manager = new Manager("John Doe", "123", "foo@example.com", "100");
      expect(manager.getOfficeNumber()).toBe("100");
    });
  });
});
