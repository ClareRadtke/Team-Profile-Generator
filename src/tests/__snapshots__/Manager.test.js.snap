// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manager should create a new Manager instance 1`] = `
Manager {
  "email": "foo@example.com",
  "id": "123",
  "name": "John Doe",
  "officeNumber": "100",
}
`;
